File: Dominant Pairs [12-April-2023]/solution.cpp
//  >> C++ CODE <<


int dominantPairs(int n,vector<int> &arr){

        sort(arr.begin(), arr.begin() + n/2);
        sort(arr.begin() + n/2, arr.end());
    
        int i = 0, j = n/2, ans = 0;
    
        while (i < n/2 && j < n) {
            if (arr[i] >= 5 * arr[j]) {
                ans += (n/2 - i);
                j++;
            }
            else {
                i++;
            }
        }
    
        return ans;
    } 

File: Dominant Pairs [12-April-2023]/Solution.java
//    >> JAVA CODE <<
 

public static int dominantPairs(int n, int[] arr) {
    
         Arrays.sort(arr,0,n/2);
         Arrays.sort(arr,n/2,n);
         
         int i=0,j=n/2;
         int ans=0;
         
         
         while(i<n/2 && j<n)
         {
             if(arr[i]>=5*arr[j])
             {
                 ans+=(n/2-i);
                 j++;
             }
             else
             {
                 i++;
             }
         }
         
         return ans;
    }